src/transformers/generation/utils.py

```diff
@@ -1280,6 +1281,28 @@ def _prepare_generation_config(
 
         return generation_config, model_kwargs
 
+    def _get_initial_cache_position(self, input_ids, model_kwargs):
+        """Calculates `cache_position` for the pre-fill stage based on `input_ids` and optionally past length"""
+        past_length = 0
+        if "past_key_values" in model_kwargs:
+            if isinstance(model_kwargs["past_key_values"], Cache):
+                past_length = model_kwargs["past_key_values"].get_seq_length()
+            elif "gptbigcode" in self.__class__.__name__.lower() or (
+                self.config.architectures is not None and "gptbigcode" in self.config.architectures[0].lower()
+            ):
+                if self.config.multi_query:
+                    past_length = model_kwargs["past_key_values"][0].shape[1]
+                else:
+                    past_length = model_kwargs["past_key_values"][0].shape[2]
```

Comment:
Rather than have this model specific code in the mixin side - it would be cleaner for inidividual models to implement their own `_get_initial_cache_position` wherever necessary

Reply:
Hmm, This one is special only fot GPT bigcode, but I think we can make special method in bigcode's modeling and fallback to the general method in all other model types.

Reply:
We can remove the gptbigcode logic here now

Reply:
oppps, removed